public/app/features/trails/DataTrailsHistory.tsx

```diff
@@ -1,43 +1,66 @@
 import { css, cx } from '@emotion/css';
 import { useMemo } from 'react';
 
-import { GrafanaTheme2 } from '@grafana/data';
+import { getTimeZoneInfo, GrafanaTheme2, InternalTimeZones, TIME_FORMAT } from '@grafana/data';
+import { convertRawToRange } from '@grafana/data/src/datetime/rangeutil';
 import {
-  SceneObjectState,
-  SceneObjectBase,
+  getUrlSyncManager,
   SceneComponentProps,
-  SceneVariableValueChangedEvent,
+  SceneObjectBase,
+  SceneObjectState,
   SceneObjectStateChangedEvent,
+  SceneObjectUrlValue,
+  SceneObjectUrlValues,
   SceneTimeRange,
   sceneUtils,
+  SceneVariableValueChangedEvent,
 } from '@grafana/scenes';
-import { useStyles2, Tooltip, Stack } from '@grafana/ui';
+import { Stack, Tooltip, useStyles2 } from '@grafana/ui';
 
 import { DataTrail, DataTrailState, getTopSceneFor } from './DataTrail';
+import { SerializedTrailHistory } from './TrailStore/TrailStore';
 import { reportExploreMetrics } from './interactions';
 import { VAR_FILTERS } from './shared';
 import { getTrailFor, isSceneTimeRangeState } from './utils';
 
 export interface DataTrailsHistoryState extends SceneObjectState {
   currentStep: number;
   steps: DataTrailHistoryStep[];
+  filtersApplied: string[];
 }
 
 export function isDataTrailsHistoryState(state: SceneObjectState): state is DataTrailsHistoryState {
   return 'currentStep' in state && 'steps' in state;
 }
 
+export function isDataTrailHistoryFilter(filter?: SceneObjectUrlValue): filter is string[] {
+  return !!filter;
+}
+
+const isString = (value: unknown): value is string => typeof value === 'string';
+
 export interface DataTrailHistoryStep {
   description: string;
+  detail: string;
   type: TrailStepType;
   trailState: DataTrailState;
   parentIndex: number;
 }
 
 export type TrailStepType = 'filters' | 'time' | 'metric' | 'start';
+
+const filterSubst = ` $2 `;
+const filterPipeRegex = /(\|)(=|=~|!=|>|<|!~)(\|)/g;
+const stepDescriptionMap: Record<TrailStepType, string> = {
+  start: 'Start of history',
+  metric: 'Metric selected',
+  filters: 'Filter applied',
+  time: 'Time range changed',
```

Comment:
I think we need to add cases for metric removed and filter remved here for those specific message.

Reply:
I'll do that in the second iteration. Please see this https://github.com/grafana/grafana/pull/90825#issuecomment-2245503932

Also metric selected message should show the selected metric as well. I think there is a little bug that is not showing it.

Reply:
The image I shared was for the workflow "Select a new metric," so not a bug, but applying the code to a new case 😃 

1. Click "Select a new metric"
![Screenshot 2024-07-24 at 8 29 55 AM](https://github.com/user-attachments/assets/5f2a4633-9f78-41e5-a4d3-82d212c3fc01)

2. The tooltip shows "Metric selected"
![Screenshot 2024-07-24 at 8 30 10 AM](https://github.com/user-attachments/assets/25891bb8-f8ad-40a3-8915-0345d656f24e)

Reply:
I suggested this copy (see comment) if we cannot separate the select a new metric from other metric select actions.
https://github.com/grafana/grafana/pull/90825#issuecomment-2245602939